python/lsst/analysis/tools/actions/plot/histPlot.py

```diff
@@ -216,56 +270,118 @@ def _assignColors(self):
+
+        ax.legend(fontsize=legend_font_size, loc="upper left", frameon=False)
         ax.set_xlim(panel_range)
-        ax.set_xlabel(self.panels[panel].label)
+        # The following accommodates spacing for ranges with large numbers
+        # but small-ish dynamic range (example use case: RA 300-301).
+        if ncols > 1 and max(np.abs(panel_range)) >= 100 and (panel_range[1] - panel_range[0]) < 5:
+            ax.xaxis.set_major_formatter("{x:.2f}")
+            ax.tick_params(axis="x", labelrotation=25, pad=-1)
+        ax.set_xlabel(self.panels[panel].label, fontsize=label_font_size)
+        y_label = "Normalized (PDF)" if self.panels[panel].histDensity else "Frequency"
+        ax.set_ylabel(y_label, fontsize=label_font_size)
         ax.set_yscale(self.panels[panel].yscale)
-        ax.tick_params(labelsize=7)
+        ax.tick_params(labelsize=max(5, label_font_size - 2))
         # add a buffer to the top of the plot to allow headspace for labels
         ylims = list(ax.get_ylim())
         if ax.get_yscale() == "log":
             ylims[1] = 10 ** (np.log10(ylims[1]) * 1.1)
         else:
             ylims[1] *= 1.1
         ax.set_ylim(ylims[0], ylims[1])
+
+        # Draw a vertical line at expected value, if given.  If histDensity
+        # is True, also plot a reference PDF with mean = expected value and
+        # sigma = 1 for reference.
+        if self.panels[panel].expectedValue is not None:
+            ax2 = ax.twinx()
+            ax2.axis("off")
+            ax2.set_xlim(ax.get_xlim())
+            ax2.set_ylim(ax.get_ylim())
+
+            if self.panels[panel].histDensity:
+                expected_label = None
+            else:
+                expected_label = "${{\\mu_{{expected}}}}$: {}".format(self.panels[panel].expectedValue)
+            ax2.axvline(
+                self.panels[panel].expectedValue, ls="-", lw=1, c="black", zorder=0, label=expected_label
+            )
+            if self.panels[panel].histDensity:
+                ref_x = np.arange(panel_range[0], panel_range[1], (panel_range[1] - panel_range[0]) / 100.0)
+                ref_mean = self.panels[panel].expectedValue
+                ref_std = 1.0
+                ref_y = (
+                    1.0
+                    / (ref_std * np.sqrt(2.0 * np.pi))
+                    * np.exp(-((ref_x - ref_mean) ** 2) / (2.0 * ref_std**2))
+                )
+                ax2.fill_between(ref_x, ref_y, alpha=0.1, color="black", label="P$_{{norm}}(0,1)$", zorder=-1)
+                # Make sure the y-axis extends beyond the data plotted and that
+                # both axes y-ranges are in sync.
+                y_max = max(max(ref_y), ax2.get_ylim()[1])
+                if ax2.get_ylim()[1] < 1.05 * y_max:
+                    ax.set_ylim(ax.get_ylim()[0], 1.05 * y_max)
+                    ax2.set_ylim(ax.get_ylim())
+            ax2.legend(fontsize=legend_font_size, handlelength=1.5, loc="upper right", frameon=False)
```

Comment:
Would it be possible to move everything below `if self.panels[panel].expectedValue is not None:` into its own separate private method? Something like `_addReferenceLines` or similar? I think that would help the legibility of this method a fair amount.